Lib/test/test_threading.py

```diff
@@ -2103,6 +2103,63 @@ def test__all__(self):
         support.check__all__(self, threading, ('threading', '_thread'),
                              extra=extra, not_exported=not_exported)
 
+    def test_set_name(self):
+        try:
+            get_name = _thread._get_name
+            _thread.set_name
+        except AttributeError:
+            self.skipTest("need thread._get_name() and thread.set_name()")
+
+        def work():
+            nonlocal work_name
+            work_name = get_name()
+
+        # set_name() limit in bytes
+        truncate = getattr(_thread, "_NAME_MAXLEN", None)
+        limit = truncate or 100
+
+        def create_test(name):
+            if sys.platform.startswith("solaris"):
+                encoding = "utf-8"
+            else:
+                encoding = sys.getfilesystemencoding()
+            encoded = name.encode(encoding, "replace")
+            if truncate is not None:
+                expected = os.fsdecode(encoded[:truncate])
+            else:
+                expected = os.fsdecode(encoded)
```

Comment:
It works with None too.
```suggestion
            expected = os.fsdecode(encoded[:truncate])
```

Reply:
Correct, but I prefer an explicit if/else :-)